src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -546,7 +480,9 @@ public Response getVersionFiles(@Context ContainerRequestContext crc,
             } catch (IllegalArgumentException e) {
                 return badRequest(BundleUtil.getStringFromBundle("datasets.api.version.files.invalid.access.status", List.of(accessStatus)));
             }
-            return ok(jsonFileMetadatas(datasetVersionFilesServiceBean.getFileMetadatas(datasetVersion, limit, offset, fileSearchCriteria, fileOrderCriteria)));
+            // TODO: should we count the total every time or only when offset = 0?
```

Comment:
Since the goal is to homogenize pagination endpoints (search API and files API), I would count and return it every time, as the search endpoint does. In this way we also guarantee that we always get the actual total count, even if there have been deletions during user paging (Not very likely).

The only reason I would consider doing it only when _offset=0_ would be because of a performance bottleneck in the count query, but I don't think that's the case.